src/core/PluginFactory.cpp

```diff
@@ -150,42 +157,79 @@ void PluginFactory::discoverPlugins()
 	for (const QFileInfo& file : files)
 	{
 		auto library = std::make_shared<QLibrary>(file.absoluteFilePath());
-
 		if (! library->load()) {
 			m_errors[file.baseName()] = library->errorString();
 			qWarning("%s", library->errorString().toLocal8Bit().data());
 			continue;
 		}
-		if (library->resolve("lmms_plugin_main") == nullptr) {
-			continue;
-		}
 
-		QString descriptorName = file.baseName() + "_plugin_descriptor";
-		if( descriptorName.left(3) == "lib" )
+		Plugin::Descriptor* pluginDescriptor = nullptr;
+		if (library->resolve("lmms_plugin_main"))
 		{
-			descriptorName = descriptorName.mid(3);
+			QString descriptorName = file.baseName() + "_plugin_descriptor";
+			if( descriptorName.left(3) == "lib" )
+			{
+				descriptorName = descriptorName.mid(3);
+			}
+
+			pluginDescriptor = reinterpret_cast<Plugin::Descriptor*>(library->resolve(descriptorName.toUtf8().constData()));
```

Comment:
you would have OOB read if `sizeof(Plugin::Descriptor)` is greater than the actual returned size.

Reply:
Can you please elaborate?

Reply:
Since you directly reads from a memory buffer without checking its actual length, you could crash in here if the returned symbol is shorter than Descriptor. (It's minor and should not happen, but also isn't a good practice).

Reply:
I think this can only happen if plugin and LMMS have different opinions on what `Plugin::Descriptor` is, i.e. if they are linked to different LMMS core versions?

Do you have any suggestions how to improve that?

Reply:
Resolving, as this can't really be fixed, and it's not really part of this PR.